src/components/App/App.js

```diff
@@ -42,19 +43,30 @@ class App extends Component {
         <div className="landing-img">
           <img src="https://images.unsplash.com/photo-1489599849927-2ee91cede3ba?ixid=MnwxMjA3fDB8MHxzZWFyY2h8MXx8Y2luZW1hfGVufDB8fDB8fA%3D%3D&ixlib=rb-1.2.1&w=1000&q=80" alt="movie icon" />
         </div>
-        { this.state.error &&
-          <h1>{this.state.error}</h1>
-        }
         {!this.state.movies.length &&
           !this.state.error &&
           <h1>Loading...</h1>
         }
-        {!this.state.currentMovie &&
-          <AllMovies movieData={this.state.movies} handleClick={this.handleClick}/>
-        }
-        {this.state.currentMovie &&
-          <MovieInfo currentMovieInfo={this.state.currentMovie} returnHome={this.returnHome} />
-        }
+        <Switch>
+          <Route
+          exact path='/movieInfo/:id'
+          render={({ match }) => {
```

Comment:
Glad we figured this out how to use match so that our page correctly loads when we click the back button!